src/service/reports.js

```diff
@@ -0,0 +1,75 @@
+const { userModel, reportModel } = require("../db/mongo");
+const logger = require("../modules/logger");
+
+const reportPopulateOption = [
+  {
+    path: "creatorId",
+    select: "-_id user",
+    populate: { path: "user", select: "_id id name nickname profileImageUrl" },
+  },
+  {
+    path: "reportedId",
+    select: "-_id user",
+    populate: { path: "user", select: "_id id name nickname profileImageUrl" },
+  },
+];
+
+const createHandler = async (req, res) => {
+  try {
+    const { reportedId, type, etcDetail, time } = req.body;
+    const user = await userModel.findOne({ id: req.userId });
+    const creatorId = user._id;
+
+    const reported = await userModel.findById(reportedId);
+    if (!reported) {
+      return res.status(400).json({
+        error: "User/report: no corresponding user",
+      });
+    }
+
+    const report = new reportModel({
+      creatorId: creatorId,
+      reportedId: reportedId,
+      type: type,
+      etcDetail: etcDetail,
+      time: time,
+    });
+
+    await report.save();
+    res.status(200).send("User/report : report successful");
+  } catch (err) {
+    logger.error(err);
+    res.status(500).json({
+      error: "User/report : internal server error",
+    });
+  }
+};
+
+const searchByUserHandler = async (req, res) => {
+  try {
+    // 해당 user가 신고한 사람인지, 신고 받은 사람인지 기준으로 신고를 분리해서 응답을 전송합니다.
+    const response = {
+      reporting: [],
+      reported: [],
+    };
+    response.reporting = await reportModel
+      .find({ creatorId: req.userId })
+      .limit(1000)
+      .populate(reportPopulateOption);
+    response.reported = await reportModel
+      .find({ reportedId: req.userId })
+      .limit(1000)
+      .populate(reportPopulateOption);
```

Comment:
```suggestion
    const response = {
      reporting: await reportModel
        .find({ creatorId: req.userId })
        .limit(1000)
        .populate(reportPopulateOption),
      reported: await reportModel
        .find({ reportedId: req.userId })
        .limit(1000)
        .populate(reportPopulateOption),
    }
```
구지 [], []로 초기화 해서 사용할 필요 없지 않을까요?